libr/io/io.c

```diff
@@ -1007,16 +1007,21 @@ static ut8 *r_io_desc_read(RIO *io, RIODesc *desc, ut64 *out_sz) {
 		return NULL;
 	}
 	if (io->maxalloc && *out_sz > io->maxalloc) {
-		eprintf ("WARNING: File is greater than 0x%"PFMT64x" bytes.\nTry setting " \
-			"R_IO_MAX_ALLOC environment variable with the desired max " \
-			"allocation bytes.\n", io->maxalloc);
-		return NULL;
+		eprintf ("WARNING: File is greater than 0x%"PFMT64x" bytes.\nAllocating R_IO_MAX_ALLOC set as the environment variable.\n", io->maxalloc);
+		*out_sz = io->maxalloc;
 	}
 
 	buf = malloc (*out_sz);
 	if (!buf) {
-		eprintf ("Cannot allocate %" PFMT64d " bytes\n", *out_sz);
-		return NULL;
+		if (*out_sz > R_IO_MAX_ALLOC) {
+			eprintf ("Failed to allocate %" PFMT64u " bytes.\nAllocating %" PFMT64u " bytes.\n", *out_sz, R_IO_MAX_ALLOC);
```

Comment:
Ok good!

> On 02 Jul 2016, at 12:43, Rakholiya Jenish <notifications@github.com> wrote:
> 
> In libr/io/io.c:
> 
> >  	}
> >  
> >  	buf = malloc (*out_sz);
> >  	if (!buf) {
> > -		eprintf ("Cannot allocate %" PFMT64d " bytes\n", *out_sz);
> > -		return NULL;
> > +		if (*out_sz > R_IO_MAX_ALLOC) {
> > +			eprintf ("Failed to allocate %" PFMT64u " bytes.\nAllocating %" PFMT64u " bytes.\n", *out_sz, R_IO_MAX_ALLOC);
> oh. Sure. I will update it. Currently fixing the failing test case.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>